tests/test_frontdoor.py

```diff
@@ -2112,6 +2122,56 @@ def test_epc_page_shows_epc_info():
     assert page.has_one("p:contains('23 July 2010')")
 
 
+@unittest.mock.patch("help_to_heat.frontdoor.interface.EPCApi", MockRecommendationsNotFoundEPCApi)
+def test_on_recommendations_not_found_response_recommendations_are_empty_list():
+    _get_empty_recommendations_journey()
+
+
+@unittest.mock.patch("help_to_heat.frontdoor.interface.EPCApi", MockRecommendationsInternalServerErrorEPCApi)
+def test_on_recommendations_internal_server_error_response_recommendations_are_empty_list():
```

Comment:
Would it be easy to have a test asserting specifically that we retry? e.g. mock an error the first time and then resolve the second?

Reply:
don't think so; the EPC API classes are stateless (theyre reconstructed each time theyre used) & using some sort of global state could have issues with this state leaking out of the test

Reply:
I was imagining something like
```
class MockRecommendationsTransientInternalServerErrorEPCApi(MockEPCApi):
    number_of_calls = 0
    def get_epc_recommendations(self, lmk_key):
        number_of_calls++
        if number_of_calls == 1:
            raise InternalServerErrorRequestException()

        return something normal
```

Would that work? We're retrying calls on the same instance.

Reply:
ah yes it can store an epc api per request, have added one now